nvmeadm/src/nvmf_discovery.rs

```diff
@@ -73,6 +73,19 @@ impl fmt::Display for TrType {
     }
 }
 
+impl FromStr for TrType {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "rdma" => Ok(Self::rdma),
+            "tcp"  => Ok(Self::tcp),
+            "fc"   => Ok(Self::fc),
+            _ => Err(format!("Invalid transport string for TrType")),
```

Comment:
Rust lint fails. Suggesting to use .to_string() instead of format!

Reply:
Modified. Linter passes now.